src/snowflake/connector/result_batch.py

```diff
@@ -519,18 +556,57 @@ def from_data(
 
         return new_chunk
 
-    def create_iter(
-        self, **kwargs
+    def _create_iter(
+        self, iter_unit: IterUnit
     ) -> Union[
         Iterator[Union[Dict, Exception]],
         Iterator[Union[Tuple, Exception]],
         Iterator[Table],
     ]:
-        iter_unit = kwargs.pop("iter_unit", ROW_UNIT)
+        """Create an iterator for the ResultBatch. Used by get_arrow_iter."""
         if self._local:
             return self._from_data(self._data, iter_unit)
         response = self._download()
         with TimerContextManager() as load_metric:
             loaded_data = self._load(response, iter_unit)
         self._metrics[DownloadMetrics.load.value] = load_metric.get_timing_millis()
         return loaded_data
+
+    def create_iter(
+        self, **kwargs
+    ) -> Union[
+        Iterator[Union[Dict, Exception]],
+        Iterator[Union[Tuple, Exception]],
+        Iterator[Table],
+    ]:
+        """The interface used by ResultSet to create an iterator for this ResultBatch."""
+        iter_unit: IterUnit = kwargs.pop("iter_unit", IterUnit.ROW_UNIT)
+        if iter_unit == IterUnit.TABLE_UNIT:
+            structure = kwargs.pop("structure", "pandas")
```

Comment:
I chose to name this `structure` but please feel free to provide any suggestions